docs/tutorials/multimodal/advanced_topics/tensorrt.ipynb

```diff
@@ -0,0 +1,411 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "779ace1c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from autogluon.multimodal import MultiModalPredictor\n",
+    "hyperparameters = {\n",
+    "    \"optimization.max_epochs\": 2,\n",
+    "    \"model.names\": [\"numerical_mlp\", \"categorical_mlp\", \"timm_image\", \"hf_text\", \"fusion_mlp\"],\n",
+    "    \"model.timm_image.checkpoint_name\": \"mobilenetv3_small_100\",\n",
+    "    \"model.hf_text.checkpoint_name\": \"google/electra-small-discriminator\",\n",
+    "    \n",
+    "}\n",
+    "predictor = MultiModalPredictor(label=label_col).fit(\n",
+    "    train_data=train_data,\n",
+    "    hyperparameters=hyperparameters,\n",
+    "    time_limit=120, # seconds\n",
+    ")\n",
+    "\n",
+    "clear_output()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ae8e9de8",
+   "metadata": {},
+   "source": [
+    "Under the hood, AutoMM automatically infers the problem type (classification or regression), detects the data modalities, selects the related models from the multimodal model pools, and trains the selected models. If multiple backbones are available, AutoMM appends a late-fusion model (MLP or transformer) on top of them.\n",
+    "\n",
+    "## Prediction with default PyTorch module\n",
+    "Given a multimodal dataframe without the label column, we can predict the labels.\n",
+    "\n",
+    "Note that we would use a small sample of test data here for benchmarking. Later, we would evaluate over the whole test dataset to assess accuracy loss."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0b0bc369",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "batch_size = 2\n",
+    "n_trails = 10\n",
+    "sample = test_data.head(batch_size)\n",
+    "\n",
+    "# Use first prediction for initialization (e.g., allocating memory)\n",
+    "y_pred = predictor.predict_proba(sample)\n",
+    "\n",
+    "pred_time = []\n",
+    "for _ in range(n_trails):\n",
+    "    tic = time.time()\n",
+    "    y_pred = predictor.predict_proba(sample)\n",
+    "    elapsed = time.time()-tic\n",
+    "    pred_time.append(elapsed)\n",
+    "    print(f\"elapsed (pytorch): {elapsed*1000:.1f} ms (batch_size={batch_size})\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "912bdb80",
+   "metadata": {},
+   "source": [
+    "## Prediction with TensorRT module\n",
+    "First, we need to export the module to ONNX, in order to use TensorrtExecutionProvider in onnxruntime for prediction."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0448f763",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model_path = predictor.path\n",
+    "trt_predictor = MultiModalPredictor.load(path=model_path)\n",
+    "trt_predictor.optimize_for_inference()\n",
+    "\n",
+    "# Agagin, use first prediction for initialization (e.g., allocating memory)\n",
```

Comment:
Typo `agagin`?

Reply:
Maybe give more explanations of using the first prediction for initialization. Otherwise, users may wonder why this is necessary.

Reply:
> Typo agagin?

Nice catch. Fixed typo.

> Maybe give more explanations of using the first prediction for initialization. Otherwise, users may wonder why this is necessary.

This is indicated with an example being `allocating memory`

Reply:
The first time calling model forward takes more time than the following calls. I'm not sure whether we can explain this observation as model initialization since Pytorch uses dynamic graph and eager execution.

Reply:
What if the batch size is changeable during predictions?

Reply:
in my observation, it would automatically re-compile when batch_size dimension is larger than initialization.

Reply:
Any Pytorch documentation regarding this phenomenon?

Reply:
The re-compile only happens to tensorrt backend in onnxruntime, not directly related to pytorch.

This behavior is not well documented, it may related to [Shape Inference for TensorRT Subgraphs](https://onnxruntime.ai/docs/execution-providers/TensorRT-ExecutionProvider.html#shape-inference-for-tensorrt-subgraphs)

Reply:
The above pytorch module also uses the first prediction as initialization. That's why we use `Again` here. So, this behavior exists for both pytorch and onnxruntime/tensorrt?

Reply:
For pytorch, it is used for memory allocation; for onnxruntime/tensorrt, it is used for 1) fair comparison, 2) model compilation.